campaignion_activity/campaignion_activity.module

```diff
@@ -16,22 +19,37 @@ function campaignion_activity_views_api() {
   );
 }
 
-if (module_exists('webform')) {
-  require_once dirname(__FILE__) . '/modules/webform.php';
-  if (module_exists('webform_paymethod_select')) {
-    require_once dirname(__FILE__) . '/modules/webform_paymethod_select.php';
+/**
+ * Implements hook_campaignion_action_taken().
+ */
+function campaignion_activity_campaignion_action_taken($node, Submission $submission, int $when) {
+  if (!($contact = $submission->contact ?? NULL)) {
+    return;
+  }
+
+  $activity = WebformSubmission::fromSubmission($submission, $contact);
+  $activity->contact_id = $contact->contact_id;
+  $activity->confirmed = $activity->confirmed ?? ($submission->confirmed ? $when : NULL);
+
+  if ($payments = $submission->payments ?? NULL) {
+    foreach ($payments as $payment) {
+      if (payment_status_is_or_has_ancestor($payment->getStatus()->status, PAYMENT_STATUS_SUCCESS)) {
+        $activity = new WebformPayment($activity);
```

Comment:
Don’t we want a WebformSubmission and a WebformPayment activity?

Reply:
This doesn’t create another activity it extends the existing one. So there will be one activity with both entries in `campaignion_activity_webform` and `campaignion_activity_payment`.

Reply:
OK